src/online/vatsim.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpService, Injectable, Logger } from '@nestjs/common';
+import { CacheService } from '../cache/cache.service';
+import { ATCInfo } from './atc-info.class';
+import { OnlineService } from './online.service';
+
+@Injectable()
+export class VatsimService {
+  private readonly logger = new Logger(VatsimService.name);
+
+  constructor(
+    private http: HttpService,
+    private readonly cache: CacheService,
+    private onlineService: OnlineService
+  ) {}
+
+  public async getControllers(): Promise<ATCInfo[]> {
+      const fetch = require('node-fetch');
+      const data: any = await (
+          await fetch('https://data.vatsim.net/v3/vatsim-data.json')
+      ).json();
+
+      const transceivers: any = await (
+          await fetch('https://data.vatsim.net/v3/transceivers-data.json')
+      ).json();
```

Comment:
Use the Nest.js HttpService instead of fetch.
These requests should also be cached. This exact same data is being used in the ATIS API already: https://github.com/flybywiresim/api/blob/5344075ae462a7c6dfb6936c4ff7da800fb0a939/src/atis/atis.service.ts#L71-#L87
Try and combine them for better caching.

Reply:
Updated